Mathlib/Algebra/GroupRingAction/Basic.lean

```diff
@@ -0,0 +1,105 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+
+! This file was ported from Lean 3 source module algebra.group_ring_action.basic
+! leanprover-community/mathlib commit 207cfac9fcd06138865b5d04f7091e46d9320432
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Ring.Equiv
+import Mathlib.Algebra.Field.Defs
+import Mathlib.GroupTheory.GroupAction.Group
+
+/-!
+# Group action on rings
+
+This file defines the typeclass of monoid acting on semirings `MulSemiringAction M R`,
+and the corresponding typeclass of invariant subrings.
+
+Note that `Algebra` does not satisfy the axioms of `MulSemiringAction`.
+
+## Implementation notes
+
+There is no separate typeclass for group acting on rings, group acting on fields, etc.
+They are all grouped under `MulSemiringAction`.
+
+## Tags
+
+group action, invariant subring
+
+-/
+
+
+universe u v
+
+/-- Typeclass for multiplicative actions by monoids on semirings.
+
+This combines `DistribMulAction` with `MulDistribMulAction`. -/
+class MulSemiringAction (M : Type u) (R : Type v) [Monoid M] [Semiring R] extends
+  DistribMulAction M R where
+  /-- Multipliying `1` by a scalar gives `1` -/
+  smul_one : ∀ g : M, (g • (1 : R) : R) = 1
+  /-- Scalara multiplication distributes across multiplication -/
+  smul_mul : ∀ (g : M) (x y : R), g • (x * y) = g • x * g • y
+#align mul_semiring_action MulSemiringAction
+
+section Semiring
+
+variable (M N G : Type _) [Monoid M] [Monoid N] [Group G]
+
+variable (A R S F : Type v) [AddMonoid A] [Semiring R] [CommSemiring S] [DivisionRing F]
+
+-- note we could not use `extends` since these typeclasses are made with `old_structure_cmd`
```

Comment:
What's going on with this comment -- I guess it's just not relevant anymore?

Reply:
I would leave it there. This instance is defined like this because `MulSemiringAction` does not extend `MulDistribMulAction`. Maybe the definition of `MulSemiringAction should be changed to use `extends`